### ISSUE TITLE
UIDocumentInteractionController shows blank previews in Intune app on iOS 16.1

### ISSUE THREAD

**Issue by dtatcbrain** [msintuneappsdk/ms-intune-app-sdk-ios#285]
**Describe the bug:**
We have an Intune app where we use a UIDocumentInteractionController to preview local documents of various types (PDF, DOCX etc). After upgrading to iOS 16.1 as well as iPadOS 16.1, users are experiencing blank previews, ie. documents are not displayed. Not every time, but often. It seems to happen most often with document types other than PDF, but does also happen for PDFs.

**To Reproduce**
Files are saved locally, and then we call:

```
UIDocumentInteractionController *previewController = [UIDocumentInteractionController interactionControllerWithURL:fileUrl];
previewController.delegate = self;
previewController.name = displayFileName;
BOOL presented = [previewController presentPreviewAnimated:YES];
```

Example fileUrl: file:///var/mobile/Containers/Data/Application/5C1ECF0C-CE3F-40B3-BF74-17CE1F7BA1A4/Library/Caches/documentcache/264486/Bestillinger%20%231771%20(Departementet%20(departementet%40departementet.dk))_F2_264489_0_264486_7.pdf

We have verified that the documents are saved correctly before previewing, so the issue lies in the preview itself (ie. what happens after we call UIDocumentInteractionController).

**Expected behavior:**
Documents are displayed 
![expected preview](https://user-images.githubusercontent.com/92786379/201664534-4583c013-139e-409d-9b11-f9da80ac2c17.PNG)

**Screenshots and logs:**
Screenshot: ![blank preview](https://user-images.githubusercontent.com/92786379/201664542-93ddb95e-bae9-4799-bd61-b12d364b066d.PNG)

**Smartphone (please complete the following information):**
 - Device: iPhone & iPad
 - OS: iOS/iPadOS 16.1 and 16.1.1

**Intune App SDK for iOS (please complete the following information):**
- What version of the Intune SDK are you using? Are you using the latest version?
Reproducible up to v17.1.1. Haven't tested 17.1.2, but it doesn't appear to have any fixes related to this.

- What platform is your app based in (native, Xamarin based, Cordova, etc)?
Native

- For errors post build, does the app launch without being Intune SDK integrated?
Works fine without Intune

- Who is the customer?
Municipality of Copenhagen and the Danish Ministry of Taxation, but we are also able to reproduce the issue in-house
- Do you see a trend with it only being reproduced on a specific device?
No, happens on various iPads and iPhones across environments

**Additional context:**
Note the screenshots are with an older Intune SDK - with 17.1.1, we are additionally missing the OK button in the upper left corner as reported on another issue.

**Comment by abasore** [comment]
Hi @dtatcbrain, it would be helpful in debugging this issue to have Intune logs. Since you are able to repo this consistently, could you open a service request by going to "Troubleshooting + support" -> "Help and support" -> "Intune" in Endpoint Manager? If you could collect them by calling the API mentioned [here](https://docs.microsoft.com/mem/intune/developer/app-sdk-ios#how-can-i-troubleshoot-my-app) and our support team will help direct them that would be really helpful.

**Comment by dtatcbrain** [comment]
Sure - we reproduced it and attached the logs to a support request (#‎34141619). We reproduced the issue using v15 of the Intune SDK, which is what we have in the app store-version of our app. It's a bit harder to reproduce on the latest SDK as it requires a number of attempts and the back button is missing from previews in this release, meaning we need to restart the app on each attempt. But let us know if you need logs from the latest release and we'll see if we can get around it. We have seen the issue with v17 as well.

**Comment by dtatcbrain** [comment]
Did the logs get through to you? I've attached the Intune log from the debug console here as well just in case, but I suppose you might need something from the server/cloud also.

[IntuneMAMDiagnosticFiles.txt](https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/files/10208587/IntuneMAMDiagnosticFiles.txt)

**Comment by dtatcbrain** [comment]
We did some testing and found that we can reproduce this issue with the policy "Send org data to other apps" set to "Policy managed apps", but we cannot reproduce it if set to "All apps". So it appears the problem is related to Intune fiddling with the preview to control sharing. We also noticed that a different UI appears for previews of PDF documents depending on this setting - when set to "All apps", there is a thumbnail view of the PDF pages on the left (on iPad).

We also found this old issue that similarly caused blank previews depending on the same setting: https://github.com/msintuneappsdk/ms-intune-app-sdk-ios/issues/117

**Comment by dtatcbrain** [comment]
Any chance of a fix included in the upcoming 17.2.0 release?

**Comment by abasore** [comment]
@dtatcbrain were you able to open a service request by going to "Troubleshooting + support" -> "Help and support" -> "Intune" in Endpoint Manager?

**Comment by dtatcbrain** [comment]
Sort of.

- We opened support request #34141619, but got stuck with first level support. It seems we aren't able to enroll devices properly in our test environment on the device level. Apps are however enrolled and policies are applied. Apparently this meant support was unable to process the request. I'm also not sure if they understood the need to forward logs to you guys. We ended up closing the request. 
- Our customer has everything enrolled properly and also experiences the issue - they have opened support request #34318849 and are working with their Microsoft contacts to escalate it.

**Comment by abasore** [comment]
Since this is a more involved issue, working with Microsoft support to escalate is the best way to get a resolution. In the meantime, we are working on reproducing the issue. Could you provide a video screen recording of the issue? Could you also confirm if the blank view eventually loads or if the view remains blank indefinitely?

For making a screen recording, you can go to Settings -> Control Center -> Enable Screen Recording. Then use the toggle in the control center drop down to start and stop a screen recording.

**Comment by dtatcbrain** [comment]
Certainly, I've uploaded a video at the link below. This is on an iPad running iPadOS 16.2, and the app is built with Intune 17.2.0. In the video, I open a DOCX file and a PDF file a number of times, both show up sometimes and are blank at other times. 

I have, on one or two occasions, seen the document appear after a while, but most often it does not. I just tried waiting for two minutes on a blank preview a few times, and it was still blank.

https://1drv.ms/v/s!AngW1prhWAGsjY8zUm9s2-dqRUntzg?e=ONjzcr

**Comment by dtatcbrain** [comment]
Our customer is still working to escalate the issue. Just for the record the issue is still present with iPadOS 16.3 and Intune SDK 17.2.1.

**Comment by dtatcbrain** [comment]
Our customer is now considering dropping Intune due to this issue. Have you been able to reproduce it?

**Comment by abasore** [comment]
Hello @dtatcbrain this case has been raised as an incident and is being investigated by the Intune SDK team. We are working on finding a root cause ASAP

**Comment by dtatcbrain** [comment]
Hi
On our customer's support case, Microsoft is now claiming that this issue has been resolved in Intune service release 2302. However our testing shows that the bug is still there. We're also a bit puzzled how this can be fixed in a service update rather than in the SDK. Have you guys been involved in this, and do you have a status for the issue on your end?

**Comment by abasore** [comment]
@dtatcbrain its possible the issue was routed to the wrong team. If you are still experiencing the issue could you confirm the issue reproduces on the latest SDK version and re-escalate the issue with Microsoft support?

**Comment by dtatcbrain** [comment]
This was fixed in 17.4.0 actually :)
